skglm/tests/test_estimators.py

```diff
@@ -96,12 +96,15 @@ def test_check_estimator(estimator_name):
 
 @pytest.mark.parametrize("estimator_name", dict_estimators_ours.keys())
 @pytest.mark.parametrize('X', [X, X_sparse])
-def test_estimator(estimator_name, X):
+def test_estimator(estimator_name, X, fit_intercept=False):
     if estimator_name == "GeneralizedLinearEstimator":
         pytest.skip()
     estimator_sk = dict_estimators_sk[estimator_name]
     estimator_ours = dict_estimators_ours[estimator_name]
 
+    estimator_ours.fit_intercept = fit_intercept
```

Comment:
this affects the dict of estimators (that's why I don't like this design @QB3)

if the test fails, we don't know precisely why.

@Klopfe can you add `test_intecepts.py` with a few tests of cd_solver over there (comapred with sklearn Lasso, Huber and Logreg for example), not encapsulated inside our estimators ?

Reply:
yes I will do that!